src/core/Akka.Streams/Implementation/Fusing/ActorGraphInterpreter.cs

```diff
@@ -910,7 +961,7 @@ private bool TryInit(GraphInterpreterShell shell)
         public IActorRef RegisterShell(GraphInterpreterShell shell)
         {
             _newShells.Enqueue(shell);
-            Self.Tell(ShellRegistered.Instance);
+            EnqueueToShortCircuit(ShellRegistered.Instance);
```

Comment:
Yeah I struggled with this one too, we needed two classes for the `Resume` version from the JVM. 
It's a `case class` which takes a `GraphInterpreterShell` as parameter and in some places it is used as instance with a shell but in this particular case only the "singelton" is used, for this cases we have `ShellRegistered`